### PR TITLE
Make openxla and opexla_eval backend show up in list_backends

### PR DESCRIPTION
**Pull request by JackCaoG**
The reason to keep the non-aot(openxla_eval) backend is discussed in https://github.com/pytorch/xla/issues/5430#issuecomment-1683191662.

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov

### CODE REVIEW

File: torch/_dynamo/backends/torchxla.py [hunk 1]
```diff
@@ -4,17 +4,21 @@
 from functorch.compile import make_boxed_func
 
 from ..backends.common import aot_autograd
-from ..backends.registry import register_experimental_backend as register_backend
+from ..backends.registry import (
+    register_experimental_backend as register_experimental_backend,
+)
+
+from .registry import register_backend
```
Comment (jansel) [inline torch/_dynamo/backends/torchxla.py:11]:
```suggestion
from .registry import register_experimental_backend, register_backend
```

**Review by jansel — CHANGES_REQUESTED** [review]
What happens if I pass a cpu/cuda tensor to this backend?  (For example, if I forgot `.to(xla)`.)

We should make sure users get a nice error message, since someone might try using this backend and not realize they need to move their tensors, since other backends don't require that.

File: torch/_dynamo/backends/torchxla.py (hunk 1 above)
Comment (JackCaoG) [reply torch/_dynamo/backends/torchxla.py:11]: PyTorch/XLA will throw an error complaining device is not XLA. Let me double check that error message and I can make it more explict.

**Comment by JackCaoG** [comment]
Let me add a more explicit message in bridge in a separate pytorch/xla pr.

**Comment by JackCaoG** [comment]
xla pr submitted for error message in https://github.com/pytorch/xla/pull/5499

**Review by jansel — APPROVED** [review]

**Comment by JackCaoG** [comment]
```
FAILED [1.3233s] dynamo/test_dynamic_shapes.py::DynamicShapesAotAutogradFallbackTests::test_aot_sequence_nr_dynamic_shapes - AssertionError: 'SeqN[507 chars]aten.ones_like.default|\n11|aten.expand.defaul[411 chars]t|\n' != 'SeqN[507 chars]aten.expand.default|\n11|aten.div.Scalar|\n10|[383 chars]t|\n'
Diff is 1006 characters long. Set self.maxDiff to None to see it. : To accept the new output, re-run test with envvar EXPECTTEST_ACCEPT=1 (we recommend staging/committing your changes before doing this)

To execute this test, run the following from the base repo dir:
    PYTORCH_TEST_WITH_ASAN=1 PYTORCH_TEST_WITH_UBSAN=1 python testing.py -k test_aot_sequence_nr_dynamic_shapes
```

test failure doesn't seem too relevant to this dog.

**Comment by jansel** [comment]
Seems unrelated, feel free to merge -f

**Pull request closed without merging**